### PR TITLE
add searchOnSelect to FilterSearch

### PR DESCRIPTION
**Pull request by oshi97**
Defaults to false. When set to true, will run a search when a filter is selected.

J=SLAP-2139
TEST=manual,auto

in the test site, see that it defaults to false and will not
execute a search on selection

update older unit tests, add new one

### CODE REVIEW

File: src/components/FilterSearch.tsx
Comment (nmanu1) [inline src/components/FilterSearch.tsx:99]: in the case where `searchOnSelect` is false, how will the user know which filter will be applied when they click an apply/search button? currently, the search input will update to show the filter value when it's selected. but, the user could start typing in the input again to look for other filters. in this case, should the previously selected filter be removed from headless? or do we want to display it in some way so the user knows what has already been selected and will be applied to the next search?

File: src/components/FilterSearch.tsx
Comment (nmanu1) [inline src/components/FilterSearch.tsx:100]: for `searchOnSelect=false`, would people use the `ApplyFiltersButton` to execute the search like they do for facets and static filters?

File: src/components/FilterSearch.tsx
Comment (oshi97) [reply src/components/FilterSearch.tsx:99]: Product is ok with this! They referenced zocdoc as having this exact behavior and it feeels good to me? I guess because in my brain if you have already selected a filter you're unlikely to go back and select a different one. Even if you did you probably wouldn't type a different value, then decide not to select a new value, and THEN hit the search button. Even if you did it doesn't seem terribly jank?
https://yext.slack.com/archives/C032CKFARGS/p1655220249371579

File: tests/components/FilterSearch.test.tsx
Comment (nmanu1) [inline tests/components/FilterSearch.test.tsx:189]: can we add a check that `setFilterOption` is still called?

Comment (oshi97) [reply tests/components/FilterSearch.test.tsx:197]: updated!

**Review by nmanu1 — COMMENTED** [review]
+1 (will approve after merge conflicts are resolved)

**Review by nmanu1 — APPROVED** [review]

**Pull request merged**
